### DISCUSSION TITLE
determining planar boundaries

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#632] [Ask Anything]
Hello,

I would like to determine the planar portion of a grain boundary as indicated by an ellipse in the figure below

![image](https://user-images.githubusercontent.com/73701277/102222677-0575bd80-3ee4-11eb-909f-5ec7e6800a20.png)

Could you help me to do such an operation please? My objective is to calculate the misorientation axis and angle of these planar boundaries.

Thank you a lot for your help,

Saoussen

**Comment by kilir** [comment]
Hi,
there are several ways to discriminate a "planar" part of the boundary, e.g. via a curvature threshold or an angular threshold between (higher order) grain boundary segments. However, it seems you have also drawn your ellipse based on the scale/size of this part of the grain boundary.

How would you exactly or at least in some more detail define the "planar portion of a grain boundary" without having to draw it?

Cheers,
Rüdiger

**Comment by Saoussen-ouhiba** [comment]
Thank you for your reply. The ellipse was drawn manually so that you can see what I mean by a "planar" part of a boundary. 

I would define a "planar portion of a grain boundary" as a part having a mean curvature equal to zero. 

Thanks again,

Saoussen

**Reply by kilir** [reply]
Hi Saoussen,
in that case, you need to make at least some decisions.
1) how much smoothing does your data need
2) what would you consider as the "mean curvature"; how many neighboring segments you'd consider (and this will most likely depend on a number of factors)
3) what is close enough to zero for the threshold of curvature

EDIT: handle triple junctions. Curvature isn't computed correctly for larger number of neighbors (alternatively, go grain by grain)

Try this one:

```matlab
mtexdata titanium
grains = ebsd.calcGrains;
%%
% 1) decide how much smoothing is required

smooth_iter = 5;

grainsS = smooth(grains,smooth_iter); 

% % remove tripe points since curvature uisn't computed the right way
tpi = unique(grainsS.boundary.triplePoints.boundaryId);
gbind = ones(length(grainsS.boundary),1);
gbind(tpi) = 0;
gbi=grainsS.boundary(logical(gbind));
gbi =gbi('indexed');

% 2) determine how many neighbours you'd like to consider determining the
% curvature

n_neighbours = 3;

curv = gbi.curvature(n_neighbours);

% 3) decide below which threshold your gb is considered "planar" 
plGb= curv <= 0.001;

gbplanar = gbi(plGb);

% only take a certain length
gbplanar = gbplanar(gbplanar.componentSize > 5);

plot(gbi,curv)
hold on
plot(gbplanar,'linewidth',2,'linecolor','r')
hold off

```
Cheers,
Rüdiger
